### ISSUE TITLE
False positive in Lint/UnreachableLoop

### ISSUE THREAD

**Issue by pyromaniac** [rubocop/rubocop#9919]
Hello folks, thanks for all you hard work!
Found a false-positive scenario for `Lint/UnreachableLoop` cop:

```
W: Lint/UnreachableLoop: This loop will have at most one iteration.
moo.zip { return foo }
^^^^^^^^^^^^^^^^^^^^^^
```

As far as I understand, 2 keys here: `.zip` and `return` inside of a block. Was unable to understand why does it act this way from the source unfortunately.

## Expected behavior

I believe RuboCop should not complain about such cases.

## Actual behavior

```sh
$ rubocop --debug r.rb
For /Users/pyromaniac: Default configuration from /Users/pyromaniac/.rvm/gems/ruby-2.7.3/gems/rubocop-1.18.3/config/default.yml
Inspecting 1 file
Scanning /Users/pyromaniac/r.rb
W

Offenses:

r.rb:3:1: W: Lint/UnreachableLoop: This loop will have at most one iteration.
moo.zip { return foo }
^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
Finished in 0.21687700017355382 seconds
```

## Steps to reproduce the problem

Ruby Rubocop on a file with:

```ruby
moo.zip { return foo }
```

## RuboCop version

```sh
$ [bundle exec] rubocop -V
1.18.3 (using Parser 3.0.1.1, rubocop-ast 1.7.0, running on ruby 2.7.3 x86_64-darwin20)
```

**Comment by jonas054** [comment]
The cop assumes that `zip` is a method in `Enumerable`, and thus that the method loops over the elements in its receiver (`moo`) and calls the given block for each element. Then a `return` inside the block is a problem, because it would stop the iteration. This is probably not what the author intended.

**Comment by pyromaniac** [comment]
Ah, actually you are right, didn't realize that `zip` takes a block. Then we probably can close the issue.
